### PR TITLE
[Docs] Document Bulk Processor for Java High Level REST Client

### PR DESCRIPTION
**Pull request by tlrx**
This PR adds documentation for the Bulk Processor usage with the Java High Level REST Client.

### CODE REVIEW

File: docs/java-rest/high-level/apis/bulk.asciidoc [hunk 1]
```diff
@@ -115,3 +117,95 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-errors]
...
 <2> Retrieve the failure of the failed operation
+
+[[java-rest-high-document-bulk-processor]]
+==== Bulk Processor
+
+The `BulkProcessor` simplifies the usage of the Bulk API by providing
+a utility class that allows index/update/delete operations to be
+transparently executed as they are added to the processor.
+
+In order to execute the requests, the `BulkProcessor` requires 3 things:
```
Comment (javanna) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: s/things/components ?

File: docs/java-rest/high-level/apis/bulk.asciidoc [hunk 2]
```diff
@@ -115,3 +117,95 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-errors]
...
+
+`RestHighLevelClient`:: This client is used to execute the `BulkRequest`s
+and to retrieve the `BulkResponse`s
```
Comment (javanna) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: I don't see why but BulkRequest and BulkResponse are not visualized properly

Comment (javanna) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: do we want to mention that the bulkAsync method will be called internally (as we provide it when creating the processor)

File: docs/java-rest/high-level/apis/bulk.asciidoc [hunk 3]
```diff
@@ -115,3 +117,95 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-errors]
...
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-processor-listener]
+--------------------------------------------------
+<1> Called before each execution of a `BulkRequest`, this method allows
+to know the number of operations that are going to be executed within the `BulkRequest`
+<2> Called after each execution of a `BulkRequest`, this method allows
+to know if the `BulkResponse` contains errors
+<3> Called if the `BulkRequest` failed, this method allows to know
+the failure
+
+When all requests have been added to the `BulkProcessor` and this one is
```
Comment (javanna) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: Once all requests have been added to the `BulkProcessor`, its instance needs to be closed using of the two available `close` methods.

File: docs/java-rest/high-level/apis/bulk.asciidoc [hunk 4]
```diff
@@ -115,3 +117,95 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-errors]
...
+waiting time elapsed before all the bulk requests completed
+
+The `close()` method can be used to immediately close the `BulkProcessor`:
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[bulk-processor-close]
+--------------------------------------------------
+
+Both methods flush the requests added to the processor before closing the processor
+and also forbids any new request to be added to it.
```
Comment (javanna) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: forbid (no s at the end)

**Review by javanna — APPROVED** [review]
left a few minors, LGTM otherwise. Do we also want to borrow this paragraph: https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-docs-bulk-processor.html#java-docs-bulk-processor-tests ?

File: docs/java-rest/high-level/apis/bulk.asciidoc (hunk 2 above)
Comment (tlrx) [inline docs/java-rest/high-level/apis/bulk.asciidoc]: > I don't see why but BulkRequest and BulkResponse are not visualized properly

I removed the `s` at the end of those

> do we want to mention that the bulkAsync method will be called internally (as we provide it when creating the processor)

I mentioned the method later, in the sample code where the processor is instantiated. I think we're good?

**Comment by tlrx** [comment]
> Do we also want to borrow this paragraph: https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-docs-bulk-processor.html#java-docs-bulk-processor-tests ?

I tend to think we should not add it in the high level docs. This looks like a general advice about refreshing indices before searching them in tests and it applies to all index/bulk/update/delete requests too. Also I don't really like that it indicates a large numbers of actions and 0 concurrent requests where it could be parallelized also in tests.

**Review by javanna — APPROVED** [review]
LGTM as-is thanks @tlrx

**Comment by tlrx** [comment]
Thanks @javanna

**Pull request merged**